clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3269,6 +3274,28 @@ static void handleIntelReqdSubGroupSize(Sema &S, Decl *D,
   S.AddIntelReqdSubGroupSize(D, AL, E);
 }
 
+static void handleIntelNamedSubGroupSize(Sema &S, Decl *D,
+                                         const ParsedAttr &AL) {
+  if (checkAttrMutualExclusion<IntelReqdSubGroupSizeAttr>(S, D, AL))
+    return;
+  if (checkAttrMutualExclusion<SYCLSimdAttr>(S, D, AL))
+    return;
```

Comment:
We're going to need a merge function for named subgroup size to handle these. We also need to think about what to do when the use specifies a different attribute argument on redeclarations.